### PR TITLE
CHE-2934: Add 'Link With Editor' action to the Project Explorer

### PR DESCRIPTION
**Pull request by svor**
### What does this PR do?
Linking the Project Explorer view to the active editor.

When you have multiple files open for editing, you can configure the Project Explorer to automatically bring an open file to the foreground  every time you select that open file in the Project Explorer view. 

The state of this action is stored in the user preferences. 

By default **Link wit editor** feature is off.

**How to set this option:**
This can be enabled in header of the Project Explorer view by choosing '**Link with editor**'  button.

**How it works:**
anim

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/2934

@vparfonov please review

### CODE REVIEW

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java [hunk 1]
```diff
@@ -574,6 +597,51 @@ private void restoreInitEditor(final VirtualFile file, final OpenEditorCallback
         finalizeInit(file, callback, editor, editorProvider);
     }
 
+    @Override
+    public void onSelectionChanged(SelectionChangedEvent event) {
+        final String isLinkedWithEditor = preferencesManager.getValue(LinkWithEditorAction.LINK_WITH_EDITOR);
+        if (isNullOrEmpty(isLinkedWithEditor) || !Boolean.parseBoolean(isLinkedWithEditor)) {
```
Comment (vparfonov) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java]: isNullOrEmpty(isLinkedWithEditor) useless here

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java [hunk 2]
```diff
@@ -574,6 +597,51 @@ private void restoreInitEditor(final VirtualFile file, final OpenEditorCallback
...
+            currentResource = (Resource)headObject;
+        }
+
+        EditorPartStack activePartStack = editorMultiPartStack.getActivePartStack();
+        if (currentResource == null || activePartStack == null || activeEditor == null) {
+            return;
+        }
+
+        if (!(activePart instanceof ProjectExplorerPresenter) &&
+            currentResource.getLocation().equals(activeEditor.getEditorInput().getFile().getLocation())) {
```
Comment (vparfonov) [inline ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java]: try to simplify code here

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java (hunk 1 above)
Comment (svor) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java]: Why? If value of the `LinkWithEditorAction.LINK_WITH_EDITOR` wasn't set we'll have empty result of the `preferencesManager.getValue(LinkWithEditorAction.LINK_WITH_EDITOR);`

Comment (svor) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java]: Cleaned

File: ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java (hunk 2 above)
Comment (svor) [reply ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/editor/EditorAgentImpl.java]: ok

**Review by vparfonov — APPROVED** [review]

**Pull request merged**